contracts/CPoR.sol

```diff
@@ -0,0 +1,139 @@
+                string memory symbol_,
+                uint8 decimals_,
+                address payable admin_) public {
+        // Creator of the contract is admin during initialization
+        admin = msg.sender;
+
+        // Initialize the market
+        initialize(underlying_, comptroller_, interestRateModel_, initialExchangeRateMantissa_, name_, symbol_, decimals_);
+
+        // Set the proper admin now that initialization is done
+        admin = admin_;
+    }
+
+    /**
+     * @notice User supplies assets into the market and receives cTokens in exchange
+     * @dev Overrides CErc20's mintFresh function to check the proof of reserves
+     * @dev This check can be skipped if the feed is set to the zero address
+     * @param account The address of the account which is supplying the assets
+     * @param mintAmount The amount of the underlying asset to supply
+     * @return (uint, uint) An error code (0=success, otherwise a failure, see ErrorReporter.sol), and the actual mint amount.
+     */
+    function mintFresh(address account, uint mintAmount) internal returns (uint, uint) {
+        if (feed == address(0)) {
+            return super.mintFresh(account, mintAmount);
+        }
+
+        MathError mathErr;
+        // Get the latest details from the feed
+        (,int answer,,uint updatedAt,) = AggregatorV3Interface(feed).latestRoundData();
+
+        uint oldestAllowed;
+        // Use MAX_AGE if heartbeat is not explicitly set
+        (mathErr, oldestAllowed) = subUInt(block.timestamp, heartbeat == 0 ? MAX_AGE : heartbeat);
+        if (mathErr != MathError.NO_ERROR) {
+            return (fail(Error.MATH_ERROR, FailureInfo.MINT_FEED_INVALID_TIMESTAMP), 0);
+        }
+
+        // Check that the feed's answer is updated with the heartbeat
+        if (oldestAllowed > updatedAt) {
+            return (fail(Error.TOKEN_MINT_ERROR, FailureInfo.MINT_FEED_HEARTBEAT_CHECK), 0);
+        }
+
+        // Get required info
+        uint underlyingSupply = EIP20Interface(underlying).totalSupply();
+        uint8 underlyingDecimals = EIP20Interface(underlying).decimals();
+        uint8 feedDecimals = AggregatorV3Interface(feed).decimals();
+        uint answerUint = uint(answer);
+
+        // Check that the feed and underlying token decimals are equivalent and normalize if not
+        if (underlyingDecimals < feedDecimals) {
+            (mathErr, underlyingSupply) = mulUInt(underlyingSupply, 10 ** uint(feedDecimals - underlyingDecimals));
+        } else if (feedDecimals < underlyingDecimals) {
+            (mathErr, answerUint) = mulUInt(answerUint, 10 ** uint(underlyingDecimals - feedDecimals));
+        }
+
+        if (mathErr != MathError.NO_ERROR) {
+            return (fail(Error.MATH_ERROR, FailureInfo.MINT_FEED_INVALID_DECIMALS), 0);
+        }
+
+        // Check that the supply of underlying tokens is not greater than the proof of reserves
+        if (underlyingSupply > answerUint) {
```

Comment:
```suggestion
        // Ensure that the current supply of underlying tokens is not greater than the reported reserves
        if (underlyingSupply > answerUint) {
```
I find the comment in combination with the code a bit confusing. I currently read the comment as a negation of what we are checking, as if we want it to be true, but really that's the error. I also think it would be better to use more variable names, like `reported/reportedUint` instead of generic `answer` and `currentSupply` instead of `underlying`, although I feel less strongly about the second rename.